### ISSUE TITLE
Setting theme causes coloring of UITableViewCell indicators

### ISSUE THREAD

**Issue by AnthonyOSX** [ViccAlexander/Chameleon#98]
Should be ignored on accessory buttons (as well as tint color set to the actual main color).

![image](https://cloud.githubusercontent.com/assets/6828951/12379224/cf3bf3b6-bd23-11e5-8b30-6f8d3c46af2e.png)

**Comment by jonlambert** [comment]
+1. Would love to know if there's a workaround to this.

**Comment by AnthonyOSX** [comment]
@jonlambert I'm currently getting around this problem by using the following:
```swift
UIButton.appearanceWhenContainedInInstancesOfClasses([UITableView.self]).backgroundColor = UIColor.clearColor()
```
This is, however, not a long term solution.

**Comment by sumowesley** [comment]
Outstanding tip to provide the work round the issue by @AnthonyOSX. Objective-C version below:
```
[[UIButton    appearanceWhenContainedInInstancesOfClasses:@[[UITableView class]]] setBackgroundColor:[UIColor clearColor]];
```

**Comment by ViccAlexander** [comment]
Nice! Thanks @AnthonyOSX and @sumowesley . I'll definitely add this to the to do list for the next update.

**Comment by connectdotz** [comment]
I like the theme setting very much except the UIButton background..., end up had to manually set it to Clear as many have suggested. However this does not work for the case when we wanted different background colors for the buttons in the same container... 

Giving there is no way to "unset" the UIAppearance once it is set, it would be great if we can pick and choose what entity (UIButton, UILabel etc) to apply the theme to so people can fine tune for what they need. Maybe an bitmask NS_Option or OptionSetType can be added for fine tuning?

**Comment by seieibob** [comment]
Just wanted to note here that the same thing happens to the clipboard and undo/redo buttons at the top of the keyboard on an iPad. Setting the secondary color to UIColor.clearColor works if you aren't using your secondary color for anything else.

**Comment by ashish-naik** [comment]
I am facing issue with scope bar in table view controller in iOS11 . Am setting global theme with primary as flatBlue and secondary as flatGrey.

Scope bar as shown.

<img width="372" alt="scopebar ios 11 issue" src="https://user-images.githubusercontent.com/4936420/39777959-a6b4efa0-5322-11e8-865d-b15daaf1e28d.png">

My delegate methods are a below. 
```
`func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
    
        let backgroundImage = UIImage(color: UIColor.white)

        if #available(iOS 11.0, *) {

            searchBar.showsScopeBar = true
            searchBar.sizeToFit()
            
           //this works for iOS 10 but in iOS 11 scope bar section appears blue
            //searchBar.scopeBarBackgroundImage = backgroundImage 

           //above statement  doesnt work in iOS 11 so trying below
            searchBar.setScopeBarButtonBackgroundImage(backgroundImage, for: .normal) 
            
            searchBar.setShowsCancelButton(true, animated: true)
           
            //trying to remove secondary color and add back in cancel delegate method
            Chameleon.setGlobalThemeUsingPrimaryColor(Constants.globalThemePrimaryColor,
                                                      withSecondaryColor: nil,
                                                      andContentStyle: .contrast)
        } else {
            
            searchBar.showsScopeBar = true
            searchBar.sizeToFit()
            searchBar.scopeBarBackgroundImage = backgroundImage
            searchBar.setShowsCancelButton(true, animated: true)
        }
    }`
```

setup as below

```
` override func viewDidLoad() {
        super.viewDidLoad()

        searchController.searchResultsUpdater = self
        searchController.obscuresBackgroundDuringPresentation = false
        searchController.searchBar.placeholder = "Search"
        if #available(iOS 11.0, *) {
            
            searchBar = searchController.searchBar
            searchController.searchBar.delegate = self
            
            if let textfield = searchBar.value(forKey: "searchField") as? UITextField {
                textfield.textColor = UIColor.blue
                if let backgroundview = textfield.subviews.first {
                    
                    // Background color
                    backgroundview.backgroundColor = UIColor.white
                    
                    // Rounded corner
                    backgroundview.layer.cornerRadius = 10;
                    backgroundview.clipsToBounds = true;
                    
                }
            }
            
            navigationItem.searchController = searchController
            searchController.searchBar.scopeButtonTitles = ["All", "Pending", "Closed"]

        } else {
            
            tableView.tableHeaderView = searchBar
            searchBar.scopeButtonTitles = ["All", "Pending", "Closed"]
            searchBar.delegate = self
        }
}
```
